bats/tests/containers/allowed-images.bats

```diff
@@ -0,0 +1,86 @@
+setup() {
+    load '../helpers/load'
+    RD_USE_IMAGE_ALLOW_LIST=true
+}
+
+teardown_file() {
+    load '../helpers/load'
+    run rdctl shutdown
+    assert_nothing
+}
+
+@test 'start' {
+    factory_reset
+    start_kubernetes
+    wait_for_apiserver
```

Comment:
Needs to `wait_for_container_engine` when using `RD_CONTAINER_ENGINE=moby` to make sure the `rancher-desktop` context exists before trying to pull an image.

Reply:
Done, but we aren't doing this consistently.